R/compute_metrics.R

```diff
@@ -0,0 +1,160 @@
+#' Calculate and format metrics from tuning functions
+#'
+#' @description
+#'
+#' This function computes metrics from tuning results. The arguments and
+#' output formats are closely related to those from [collect_metrics()], but
+#' this function additionally takes a `metrics` argument with a
+#' [metric set][yardstick::metric_set()] for new metrics to compute. This
+#' allows for computing new performance metrics without requiring users to
+#' re-evaluate models against resamples.
+#'
+#' Note that the [control option][control_grid()] `save_pred = TRUE` must
+#' have been supplied when generating `x`.
+#'
+#' @param x The results of a tuning function like [tune_grid()] or
+#' [fit_resamples()], generated with the control option `save_pred = TRUE`.
+#' @param metrics A [metric set][yardstick::metric_set()] of new metrics
+#' to compute. Each metric in the set must have a metric type (usually
```

Comment:
I would put the parts about the potential discordance between the types of predictions produced and what the metric set wants in a Details section. 

If there are suggestions for how `group_by()` works (or doesn't), I'd put that in Details too.